framework/areg/base/private/posix/NEUtilitiesPosix.cpp

```diff
@@ -21,79 +21,13 @@
-     *          With this method we compute the difference in seconds to balance.
-     *          The method is only for internal use and should not be used outside
-     *          of this file.
-     **/
-    inline int _internalTimeDifference( void )
-    {
-        struct timespec relTime;
-        ::clock_gettime( CLOCK_REALTIME, &relTime );
-        time_t secs = static_cast<time_t>(relTime.tv_sec);
-
-        struct tm tmGmt{0};
-        gmtime_r(&secs, &tmGmt);
-        tmGmt.tm_isdst   = -1;
-        time_t secsConv  = std::mktime( &tmGmt );
-        int64_t timeDiff = static_cast<int64_t>(secs) - static_cast<int64_t>(secsConv);
-        return static_cast<int>(timeDiff);
-    }
-
-    /**
-     * \brief   Generates a name, sets the timestamp of now.
-     * 
-     * \param   prefix          The prefix to add to name.
-     * \param   out_buffer[out] On output, this contains the result.
-     * \param   length          The length of output buffer.
-     * \param   specChar        Special character to add when generating name.
-     * \return Generated name.
-     **/
-    const char * _osGenerateName( const char * prefix, char * OUT out_buffer, int length, const char * specChar )
-    {
-        constexpr char strFormat[]   { "%s%s%08x%s%08x" };
-        
-        if ( out_buffer != nullptr )
-        {
-            const char * spec = specChar != nullptr ? specChar : NECommon::DEFAULT_SPECIAL_CHAR.data();
-            *out_buffer = '\0';
-            struct timespec now;
-            ::clock_gettime(CLOCK_MONOTONIC, &now);
-            String::formatString( out_buffer, length, strFormat
-                                , prefix != nullptr ? prefix : NEUtilities::DEFAULT_GENERATED_NAME.data()
-                                , spec
-                                , now.tv_sec
-                                , spec, now.tv_nsec);
-        }
-
-        return out_buffer;
-    }
-
-    /**
-     * \brief   Converts given time in microseconds into the time in seconds, milliseconds and microseconds.
-     * 
-     * \param   quad[in]        The time in microseconds to parse and extract.
-     * \param   outSecs[out]    On output, this contains the time in seconds.
-     * \param   outMilli[out]   On output, this contains the remaining time in milliseconds.
-     * \param   outMicro[out]   On output, this contains the remaining time in microseconds.
-     **/
-    inline void _convertMicrosecs(const TIME64 & IN quad, time_t & OUT outSecs, unsigned short & OUT outMilli, unsigned short & OUT outMicro)
-    {
-        time_t secs = static_cast<time_t>(quad / SEC_TO_MICROSECS);
-        time_t rest = static_cast<time_t>(quad % SEC_TO_MICROSECS);
-        outSecs     = secs;
-        outMilli    = static_cast<unsigned short>(rest / MILLISEC_TO_MICROSECS);
-        outMicro    = static_cast<unsigned short>(rest % MILLISEC_TO_MICROSECS);
-    }
-
-
     uint64_t _osGetTickCount( void )
```

Comment:
Please add commnet either here of in NEUtilities.cpp file where the OS specific methods are declared as `extern`.